src/software/world/robot.cpp

```diff
@@ -60,27 +46,39 @@ RobotId Robot::id() const
 
 Point Robot::position() const
 {
-    return states_.front().state().position();
+    return current_state_.position();
 }
 
 Vector Robot::velocity() const
 {
-    return states_.front().state().velocity();
+    return current_state_.velocity();
 }
 
 Angle Robot::orientation() const
 {
-    return states_.front().state().orientation();
+    return current_state_.orientation();
 }
 
 AngularVelocity Robot::angularVelocity() const
 {
-    return states_.front().state().angularVelocity();
+    return current_state_.angularVelocity();
 }
 
-RobotHistory Robot::getPreviousStates() const
+bool Robot::isNearDribbler(const Point &test_point) const
 {
-    return states_;
+    Vector vector_to_test_point                     = test_point - position();
+    static const double POSSESSION_THRESHOLD_METERS = ROBOT_MAX_RADIUS_METERS + 0.2;
```

Comment:
Is the `0.2` just from trial and error or is there a specific reason? Just curious where it came from.

Reply:
it's experimental, I've added a comment for that